catalog/queries/search/adamlink-straten.rq

```diff
@@ -14,8 +14,18 @@ WHERE {
     ?uri ?predicate ?label .
     VALUES ?predicate { skos:prefLabel skos:altLabel }
 
-    # Replace query "A B" with "A AND B", leaving queries "A AND B" or "A OR B" unchanged.
-    FILTER (<bif:contains> (?label, REPLACE(?query, "(?<!AND)(?<!OR)[[:space:]]+(?!AND)(?!OR)", " AND ", "i")))
+    {
+        FILTER(STRLEN(?booleanQuery) > 0)
+        ?uri ?predicate ?label .
+        ?label <bif:contains> ?booleanQuery .
+    } UNION {
+        # BC support.
+        BIND(?booleanQuery as ?b)
+        FILTER(!BOUND(?b))
```

Comment:
Binding `?booleanQuery` first to some other var (`?b`) was needed to make this query work. Perhaps because checking the same variable in both UNION parts trips up Comunica?

Reply:
Great find!